pkgs/development/python-modules/tokenizers/default.nix

```diff
@@ -0,0 +1,120 @@
+{ stdenv
+, rustPlatform
+, fetchFromGitHub
+, fetchurl
+, maturin
+, pip
+, pytest
+, python
+, pythonRecompileBytecodeHook
+, requests
+}:
+
+let
+  robertaVocab = fetchurl {
+    url = "https://s3.amazonaws.com/models.huggingface.co/bert/roberta-base-vocab.json";
+    sha256 = "0m86wpkfb2gdh9x9i9ng2fvwk1rva4p0s98xw996nrjxs7166zwy";
+  };
+  robertaMerges = fetchurl {
+    url = "https://s3.amazonaws.com/models.huggingface.co/bert/roberta-base-merges.txt";
+    sha256 = "1idd4rvkpqqbks51i2vjbd928inw7slij9l4r063w3y5fd3ndq8w";
+  };
+  bertVocab = fetchurl {
+    url = "https://s3.amazonaws.com/models.huggingface.co/bert/bert-base-uncased-vocab.txt";
+    sha256 = "18rq42cmqa8zanydsbzrb34xwy4l6cz1y900r4kls57cbhvyvv07";
+  };
+  openaiVocab = fetchurl {
+    url = "https://s3.amazonaws.com/models.huggingface.co/bert/openai-gpt-vocab.json";
+    sha256 = "0y40gc9bixj5rxv674br1rxmxkd3ly29p80x1596h8yywwcrpx7x";
+  };
+  openaiMerges = fetchurl {
+    url = "https://s3.amazonaws.com/models.huggingface.co/bert/openai-gpt-merges.txt";
+    sha256 = "09a754pm4djjglv3x5pkgwd6f79i2rq8ydg0f7c3q1wmwqdbba8f";
+  };
+in rustPlatform.buildRustPackage rec {
+  pname = "tokenizers";
+  version = "0.7.0";
+  name = "${python.libPrefix}-${pname}-${version}";
+
+  src = fetchFromGitHub {
+    owner = "huggingface";
+    repo = pname;
+    rev = "python-v${version}";
+    sha256 = "1f13rmqa0zy9x6hilk8j3pdly287ill6qs0xah6q7d0cj8pf5p8f";
+  };
+
+  cargoSha256 = "0729md01sk3bygbba16bc7pbyh5rnl8c1mcrkydi648zlnnly3a0";
```

Comment:
not entirely sure why, but this seems to create a different sha for python37:
```
hash mismatch in fixed-output derivation '/nix/store/47zy6vvwg0cslf1d8bx0d00x3f3azbj5-python3.7-tokenizers-0.7.0-vendor.tar.gz':
  wanted: sha256:0729md01sk3bygbba16bc7pbyh5rnl8c1mcrkydi648zlnnly3a0
  got:    sha256:0mp1ihq4977k8gmm4annqwlljbqfzmsa8rbpfqib01xfzavy8pii
cannot build derivation '/nix/store/ld60ya7x93ry9qnylgsvdnh246kgbwxf-python3.7-tokenizers-0.7.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/71ylyxpsf8yajz20d82071rz1klpqb56-env.drv': 1 dependencies couldn't be built
[8 built (1 failed), 0.0 MiB DL]
error: build of '/nix/store/71ylyxpsf8yajz20d82071rz1klpqb56-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/91342
1 package failed to build:
python37Packages.tokenizers
```

Reply:
Thanks for catching that!

`name` is used as part of the directory of vendored dependencies. Since `name` was prefixed with `python.libPrefix`, the directory name changed between Python versions, leading to a different hash of the vendored dependencies tarball.

I have now removed the `python.libPrefix` part of the name.